File: app/src/main/java/pl/orbitemobile/wspolnoty/BaseApplication.java
/*
 * Copyright (c) 2017. All Rights Reserved. Michal Jankowski orbitemobile.pl
 */
package pl.orbitemobile.wspolnoty;

import com.google.android.gms.analytics.GoogleAnalytics;
import com.google.android.gms.analytics.Tracker;

import android.app.Application;

import pl.orbitemobile.wspolnoty.R;

public class BaseApplication extends Application {
    
    private Tracker mTracker;
    
    @Override
    public void onCreate() {
        super.onCreate();
    }
    
    synchronized public Tracker getDefaultTracker() {
        if (mTracker == null) {
            GoogleAnalytics analytics = GoogleAnalytics.getInstance(this);
            mTracker = analytics.newTracker(R.xml.global_tracker);
        }
        return mTracker;
    }
}

File: app/src/main/java/pl/orbitemobile/wspolnoty/BaseApplication.kt
package pl.orbitemobile.wspolnoty

import android.app.Application
import com.google.android.gms.analytics.GoogleAnalytics
import com.google.android.gms.analytics.Tracker


class BaseApplication : Application() {

    private var mTracker: Tracker? = null

    override fun onCreate() {
        super.onCreate()
    }

    val defaultTracker: Tracker?
        @Synchronized get() {
            if (mTracker == null) {
                val analytics = GoogleAnalytics.getInstance(this)
                mTracker = analytics.newTracker(R.xml.global_tracker)
            }
            return mTracker
        }
}